kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/ColumnMapping.java

```diff
@@ -156,6 +156,111 @@ static int findMaxColumnId(StructType schema) {
     return maxColumnId;
   }
 
+  static void validateColumnIds(StructType currentSchema, StructType updatedSchema) {
+    Map<Long, String> currentFieldIdToPhysicalName = new HashMap<>();
+    for (StructField field : currentSchema.fields()) {
+      validateColumnIds(field, currentFieldIdToPhysicalName);
+    }
+
+    Map<Long, String> updatedFieldIdToPhysicalName = new HashMap<>();
+    for (StructField field : updatedSchema.fields()) {
+      validateColumnIds(field, updatedFieldIdToPhysicalName);
+    }
+
+    Set<String> dedupedPhysicalNames = new HashSet<>(updatedFieldIdToPhysicalName.values());
+    if (dedupedPhysicalNames.size() != updatedFieldIdToPhysicalName.size()) {
+      throw new IllegalArgumentException("Assigned physical names must be unique");
+    }
+
+    for (Map.Entry<Long, String> field : updatedFieldIdToPhysicalName.entrySet()) {
+      String existingPhysicalName = currentFieldIdToPhysicalName.get(field.getKey());
+      // Found an existing field, verify the physical name is preserved
+      if (existingPhysicalName != null && !existingPhysicalName.equals(field.getValue())) {
+        throw new IllegalArgumentException(
+            String.format(
+                "Existing field with id %s in current schema has physical name %s which is different from %s",
+                field.getKey(), existingPhysicalName, field.getValue()));
+      }
+    }
+  }
+
+  private static void validateColumnIds(
+      StructField field, Map<Long, String> fieldIdToPhysicalName) {
+    if (!hasPhysicalName(field)) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Column mapping mode is name and field %s is missing physical name",
+              field.getName()));
+    }
+
+    if (!hasColumnId(field)) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Column mapping mode is name and field %s is missing column id", field.getName()));
+    }
```

Comment:
Validation that physical names and column ids are defined

Reply:
need to fix the message since it applies for both column mapping modes

Reply:
Do we do this after we populate them for the non-DBI connector? We don't expect other connectors to populate their own physical names/column ids

(Referring to for table creation. For this updateSchema API I still think we should maybe make it a separate internal API and restrict it to just usage by DBI -- not opening up schema evolution for all connectors yet)